### PR TITLE
docs: Update READMEs for Blockstore and Tagstore

### PR DESCRIPTION
**Pull request by bradenmacdonald**
## Description

This repository's README was a bit out of date. This PR makes some changes to bring it up to date. More work will be needed in future PRs.

I also reinstated the Tagstore README, [per the recent discussion about it on the forum](https://discuss.openedx.org/t/guidance-needed-on-processing-course-content-text-in-course-discovery/8160).

## Test Instructions

Not sure if any testing is needed. Just look for typos and make sure the links work and the information is accurate.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @bradenmacdonald!

As a core committer in this repo, you can merge this once the pull request is approved per the [core committer reviewer requirements](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1529675973/Open+edX+Core+Committers#Reviews-and-Ownership) and according to the agreement with your edX Champion.

**Comment by bradenmacdonald** [comment]
@Agrendalath @pomegranited Would you be able to review this? Jira ticket is MNG-3295.

File: README.rst
Comment (pomegranited) [inline README.rst:62]: These instructions are for running blockstore as a service, incidentally alongside the docker devstack. Since we can now run blockstore as an app, and the docker devstack is deprecated, do we even need these instructions anymore?

But if we're keeping them, they should be updated. E.g. just below here, it says:

> Prerequisite: Have your Open edX `Devstack <https://github.com/openedx/devstack>`_ properly installed and working. Your devstack must be tracking the ``master`` branch of ``edx-platform``; using Blockstore on an older devstack release is not supported.

But above, we say that it's supported from nutmeg onwards.

The notes about running integration tests are outdated as well, and are ok to remove.

Comment (pomegranited) [reply README.rst:62]: The preface for "Using in Production" can be changed to:

> By default, blockstore is run as an app inside of Open edX. Enable it using the waffle switch [`blockstore.use_blockstore_app_api`](https://github.com/openedx/edx-platform/blob/master/openedx/core/lib/blockstore_api/config/waffle.py).
> 
> If you need to run blockstore as a separate service (e.g. for scalability or performance reasons), you can deploy blockstore in production using the `blockstore ansible role <https://github.com/openedx/configuration/tree/master/playbooks/roles/blockstore>`_.
> ...

**Review by pomegranited — CHANGES_REQUESTED** [review]
Thank you for updating this README @bradenmacdonald ! I really should have done that as part of moving blockstore into an app!

I've suggested some more changes that clarify this move and the recommended usage in production. Let me know if you want any help rewriting these?

**Comment by natabene** [comment]
@bradenmacdonald Thanks for the contribution.

File: README.rst
Comment (bradenmacdonald) [reply README.rst:59]: Yes, I definitely will - but it hasn't merged yet. Just waiting for that.

File: README.rst
Comment (bradenmacdonald) [reply README.rst]: Thanks @pomegranited ! I pushed 575c37effe9a386612e0bef06c9ed8990a2f60fa with those changes. Let me know your thoughts please :)

File: README.rst
Comment (pomegranited) [inline README.rst:142]: We can keep this section -- we still need the blockstore docker container and these instructions for running tests.

File: README.rst
Comment (pomegranited) [inline README.rst:71]: Can  you move these lines down under `Running Tests`? We should keep that section, and this info on how to run the blockstore docker container, so that people still know how to run tests.

**Review by pomegranited — CHANGES_REQUESTED** [review]

File: README.rst
Comment (bradenmacdonald) [reply README.rst:71]: Is it important to still have that way to run tests? I was under the impression that all the tests are currently run in "app mode" in the edx-platform test suite, and these are only needed to run the tests in "IDA mode".

Actually, can we soon remove "IDA mode" from edx-platform and blockstore entirely, or is there some reason we need to continue to support both options?

Comment (bradenmacdonald) [reply README.rst]: I've but those back in, in a modified form. Please review :)

**Review by pomegranited — APPROVED** [review]
:+1:  Thank you @bradenmacdonald !

- [x] I tested this by reading through the rendered markup, checking links and code snippets for accuracy.
- [x] I read through the code -- testing and setup details are still accurate.
- [x] <strike>I checked for accessibility issues</strike> N/A
- [x] Includes documentation
- [x] Commit structure follows [OEP-0051](https://github.com/edx/open-edx-proposals/pull/189)

File: README.rst
Comment (pomegranited) [reply README.rst]: Thanks for retaining those test instructions until we can get the standalone service removed.

edx.org is still running blockstore standalone in production, so we shouldn't remove it yet.

**Review by Agrendalath — APPROVED** [review]
@bradenmacdonald, thank you for preparing this, and sorry for the late reply. Looks great!

:+1:

* [x] I tested this: read the diff, checked the links
* [x] I read through the code
* [x] I checked for accessibility issues: n/a
* [x] Includes documentation

**Comment by openedx-webhooks** [comment]
@jristau1984: thought you might like to know that bradenmacdonald merged this pull request.

**Comment by openedx-webhooks** [comment]
@bradenmacdonald 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/blockstore&entry.867055334=https://github.com/openedx/blockstore/pull/204&entry.1484655318=https://github.com/bradenmacdonald&entry.752974735=2022-09-26+21:49&entry.1917517419=2022-10-03+17:54&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
